app/controllers/users_controller.rb

```diff
@@ -4,34 +4,54 @@ class UsersController < ApplicationController
   before_action :require_sign_in
   load_and_authorize_resource
   load_and_authorize_resource :vita_partner, collection: [:edit, :update], parent: false
-
+  before_action :authorize_vita_partner_updates, only: :update
   layout "admin"
 
-  def profile
-  end
+  def profile;end
 
   def index
     @users = @users.includes(:vita_partner)
   end
 
   def edit
+    @user.vita_partner_id = @user.memberships.find_by(role: "member")&.vita_partner_id
+    @user.supported_organization_ids = @user.memberships.where(role: "lead").pluck(:vita_partner_id)
   end
 
   def update
-    authorize!(:manage, @vita_partners.find(user_params[:vita_partner_id]))
-    return render :edit unless @user.update(user_params)
-
+    authorize_vita_partner_updates
+    return render :edit unless update_user
     redirect_to edit_user_path(id: @user), notice: I18n.t("general.changes_saved")
   end
 
   private
+  # To leave the view as-is but allow the ability to save as memberships, we're doing the simplest thing possible:
+  # blowing away and recreating memberships on every update.
+  def update_user
+    ApplicationRecord.transaction do
+      @user.memberships.destroy_all && @user.update!(user_params.merge(memberships_attributes: memberships))
+    end
```

Comment:
When doing something like destroying and updating related objects, it was really smart to wrap it in a transaction! Good thinking